src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenCheckedTests.cs

```diff
@@ -2367,5 +2367,391 @@ .locals init (int V_0)
+            CheckedConversionInExpressionTree_Implicit("byte", "decimal", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("byte", "float", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("byte", "double", ConvertMethod.Convert);
+
+            // short
+            CheckedConversionInExpressionTree_Implicit("short", "short", ConvertMethod.None);
+            CheckedConversionInExpressionTree_Implicit("short", "int", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("short", "long", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("short", "decimal", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("short", "float", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("short", "double", ConvertMethod.Convert);
+
+            // ushort
+            CheckedConversionInExpressionTree_Implicit("ushort", "ushort", ConvertMethod.None);
+            CheckedConversionInExpressionTree_Implicit("ushort", "int", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("ushort", "uint", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("ushort", "long", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("ushort", "ulong", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("ushort", "decimal", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("ushort", "float", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("ushort", "double", ConvertMethod.Convert);
+
+            // int
+            CheckedConversionInExpressionTree_Implicit("int", "int", ConvertMethod.None);
+            CheckedConversionInExpressionTree_Implicit("int", "long", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("int", "decimal", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("int", "float", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("int", "double", ConvertMethod.Convert);
+
+            // uint
+            CheckedConversionInExpressionTree_Implicit("uint", "uint", ConvertMethod.None);
+            CheckedConversionInExpressionTree_Implicit("uint", "long", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("uint", "ulong", ConvertMethod.ConvertChecked);
+            CheckedConversionInExpressionTree_Implicit("uint", "decimal", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("uint", "float", ConvertMethod.Convert);
+            CheckedConversionInExpressionTree_Implicit("uint", "double", ConvertMethod.Convert);
+
+            // long
+            CheckedConversionInExpressionTree_Implicit("long", "long", ConvertMethod.None);
```

Comment:
This test is testing implicit conversions, so some combinations such as `long` to `ulong` cannot be tested since there is no implicit conversion.